new-log-viewer/src/contexts/UrlContextProvider.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+    createContext,
+    useEffect,
+    useState,
+} from "react";
+
+
+interface UrlContextType {
+    setSearchParamSet: (searchParamSet: Record<string, string | null>) => void;
+    setHashParamSet: (hashParamSet: Record<string, string | null>) => void;
+    copyToClipboard: (
+        searchParamSet: Record<string, string | null>,
+        hashParamSet: Record<string, string | null>
+    ) => void;
```

Comment:
Since those methods don't really have any dependencies with the states in UrlContextProvider, do we really need to provide those as part of the UrlConext?